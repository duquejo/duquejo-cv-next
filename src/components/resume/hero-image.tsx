import Image from 'next/image';
import { cn } from '@/lib';
import { Circle, Gamepad, LoaderCircle, Music, Send } from 'lucide-react';
import avatarPicture from '../../../public/static/img/avatar_410.png';

interface Props {
  className?: string;
}

export const HeroImage = ({ className }: Props) => {
  return (
    <div data-testid="hero-image" className={cn('overflow-hidden relative', className)}>
      <Music
        className="absolute top-0 text-primary/50 animate-bounce-subtle repeat-infinite z-10"
        strokeWidth="3"
        size={54}
      />
      <Gamepad
        className="absolute bottom-0 left-7 text-primary/50 animate-wiggle z-10"
        strokeWidth="3"
        size={60}
      />
      <LoaderCircle
        className="absolute right-10 text-primary animate-spin duration-slow repeat-infinite z-10"
        strokeWidth="5"
        size={44}
      />
      <Send
        className="absolute right-8 bottom-5 text-primary animate-bounce-subtle repeat-infinite z-10"
        strokeWidth="3"
        size={50}
      />

      {/*Right*/}
      <Circle
        className="absolute right-36 top-10 text-primary/50 animate-pulse duration-slow"
        strokeWidth="5"
        size={40}
      />
      {/*Left*/}
      <Circle
        className="absolute left-20 bottom-24 text-primary/50 animate-pulse duration-slow delay-150"
        strokeWidth="5"
        size={20}
      />
      <picture className="relative">
        <Image
          className="rounded-full backdrop-blur-sm bg-primary/10 shadow-inner"
          alt="JosÃ© Miguel Duque"
          height={410}
          width={410}
          priority={true}
          quality={100}
          src={avatarPicture}
        />
      </picture>
    </div>
  );
};
