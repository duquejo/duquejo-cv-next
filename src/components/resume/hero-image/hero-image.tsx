import Image from 'next/image';
import { cn } from '@/lib';
import { Circle, Gamepad, LoaderCircle, Music, Send } from 'lucide-react';

interface Props {
  className?: string;
}

export const HeroImage = ({ className }: Props) => {
  return (
    <div data-testid="hero-image" className={cn('overflow-hidden relative', className)}>
      <Music
        className="absolute top-0 text-primary/50 animate-bounce-subtle repeat-infinite z-10"
        strokeWidth="3"
        size={64}
      />
      <Gamepad
        className="absolute bottom-0 left-7 text-primary/50 animate-wiggle z-10"
        strokeWidth="3"
        size={70}
      />
      <LoaderCircle
        className="absolute right-10 text-primary animate-spin duration-slow repeat-infinite z-10"
        strokeWidth="5"
        size={64}
      />
      <Send
        className="absolute right-8 bottom-5 text-primary animate-bounce-subtle repeat-infinite z-10"
        strokeWidth="3"
        size={50}
      />

      {/*Right*/}
      <Circle
        className="absolute right-36 top-10 text-primary/50 animate-pulse duration-slow"
        strokeWidth="5"
        size={40}
      />
      {/*Left*/}
      <Circle
        className="absolute left-20 bottom-24 text-primary/50 animate-pulse duration-slow delay-150"
        strokeWidth="5"
        size={20}
      />
      <picture className="object-cover">
        <Image
          className="rounded-full backdrop-blur-sm bg-primary/10 shadow-inner"
          alt="JosÃ© Miguel Duque"
          width="950"
          height="950"
          priority
          src="/static/img/avatar_cropped_950.png"
        />
      </picture>
    </div>
  );
};
