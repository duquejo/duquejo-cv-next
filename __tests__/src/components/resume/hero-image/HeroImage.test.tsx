import { HeroImage } from '@/components/resume/hero-image/HeroImage';
import { render } from '@testing-library/react';

describe('<HeroImage /> tests', () => {
  it('Should match the snapshot with the default args', () => {
    const { container } = render(<HeroImage />);

    expect(container).toMatchSnapshot();
  });

  it('Should add the given css classes if they are given', () => {
    const { container } = render(<HeroImage className="bg-red-500" />);

    expect(container.querySelector('picture')).toBeInTheDocument();
    expect(container.querySelector('picture')).toHaveClass('bg-red-500');
  });
});
