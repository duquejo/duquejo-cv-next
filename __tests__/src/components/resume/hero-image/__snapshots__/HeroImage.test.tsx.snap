// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HeroImage /> tests > Should match the snapshot with the default args 1`] = `
<div>
  <picture
    class="overflow-hidden relative max-w-md"
  >
    <svg
      class="lucide lucide-music absolute top-0 text-primary/50 animate-bounce-subtle repeat-infinite z-10"
      fill="none"
      height="64"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="3"
      viewBox="0 0 24 24"
      width="64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 18V5l12-2v13"
      />
      <circle
        cx="6"
        cy="18"
        r="3"
      />
      <circle
        cx="18"
        cy="16"
        r="3"
      />
    </svg>
    <svg
      class="lucide lucide-gamepad absolute bottom-0 left-5 text-primary/50 animate-wiggle z-10"
      fill="none"
      height="70"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="3"
      viewBox="0 0 24 24"
      width="70"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="6"
        x2="10"
        y1="12"
        y2="12"
      />
      <line
        x1="8"
        x2="8"
        y1="10"
        y2="14"
      />
      <line
        x1="15"
        x2="15.01"
        y1="13"
        y2="13"
      />
      <line
        x1="18"
        x2="18.01"
        y1="11"
        y2="11"
      />
      <rect
        height="12"
        rx="2"
        width="20"
        x="2"
        y="6"
      />
    </svg>
    <svg
      class="lucide lucide-loader-circle absolute right-5 text-primary animate-spin duration-slow repeat-infinite z-10"
      fill="none"
      height="64"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="5"
      viewBox="0 0 24 24"
      width="64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
    <svg
      class="lucide lucide-send absolute right-8 bottom-5 text-primary animate-bounce-subtle repeat-infinite z-10"
      fill="none"
      height="50"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="3"
      viewBox="0 0 24 24"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.536 21.686a.5.5 0 0 0 .937-.024l6.5-19a.496.496 0 0 0-.635-.635l-19 6.5a.5.5 0 0 0-.024.937l7.93 3.18a2 2 0 0 1 1.112 1.11z"
      />
      <path
        d="m21.854 2.147-10.94 10.939"
      />
    </svg>
    <svg
      class="lucide lucide-circle absolute right-32 top-10 text-primary/50 animate-pulse duration-slow"
      fill="none"
      height="40"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="5"
      viewBox="0 0 24 24"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
    <svg
      class="lucide lucide-circle absolute left-20 bottom-24 text-primary/50 animate-pulse duration-slow delay-150"
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="5"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
    <img
      alt="JosÃ© Miguel Duque"
      class="rounded-full backdrop-blur-sm bg-primary/5 shadow-inner object-cover"
      data-nimg="1"
      decoding="async"
      height="950"
      loading="lazy"
      src="/_next/image?url=%2Fstatic%2Fimg%2Favatar_cropped_950.png&w=1920&q=75"
      srcset="/_next/image?url=%2Fstatic%2Fimg%2Favatar_cropped_950.png&w=1080&q=75 1x, /_next/image?url=%2Fstatic%2Fimg%2Favatar_cropped_950.png&w=1920&q=75 2x"
      style="color: transparent;"
      width="950"
    />
  </picture>
</div>
`;
